pkgs/state-transition/src/transition.zig

```diff
@@ -1,20 +1,17 @@
 const ssz = @import("ssz");
 const std = @import("std");
-const builtin = @import("builtin");
 const Allocator = std.mem.Allocator;
 const types = @import("@zeam/types");
 pub const utils = @import("./utils.zig");
+
+const zeam_utils = @import("@zeam/utils");
+const log = zeam_utils.zeamLog;
+const getLogger = zeam_utils.getLogger;
+
 const params = @import("@zeam/params");
 
-fn log(comptime fmt: []const u8, args: anytype) !void {
-    if (builtin.target.os.tag == .freestanding) {
-        const io = @import("zkvm").io;
-        var buf: [512]u8 = undefined;
-        io.print_str(try std.fmt.bufPrint(buf[0..], fmt, args));
-    } else {
-        std.debug.print(fmt, args);
-    }
-}
+// put the active logs at debug level for now by default
+pub const StateTransitionOpts = struct { activeLogLevel: std.log.Level = std.log.Level.debug };
```

Comment:
This is ok as long as we're developing, but when we start making releases and showing our performance, we should not make debug the default. It should in fact depend on the build optimization.

Reply:
in the guest program I agree, but I like the ability to change log levels via env in the least (and with keystrokes or api call on a running client).